### ISSUE TITLE
Brochure: Re-structure cross-reference text for Resolutions, Recommendations,  etc. (French version)

### ISSUE THREAD

**Issue by manuelfuenmayor** [metanorma/metanorma-bipm#195]
In relation to https://github.com/metanorma/bipm-si-brochure/issues/189 and https://github.com/metanorma/bipm-si-brochure/issues/190

From BIPM:

Number | Page | Clause/paragraph | Comment |  
-- | -- | -- | -- | --
24 | 58 | CIPM 1967 | "CIPM recommandation 2" should be replaced by "Recommandation 2 du CIPM" | general rule, to be applied systematically

For example, this markup:
```adoc
[[cipm1967]]
[%unnumbered]
== CIPM, 1967 (((masse))) index-range:kilogramme_multiples[(((kilogramme,multiples et sous-multiples)))] index-range:multiples_kilogramme[(((multiples du kilogramme)))]

[[cipm1967r2]]
[%unnumbered]
=== {blank}

[.variant-title,type=quoted]
*Multiples et sous-multiples décimaux de l’unité de masse* (<<PV35_2>> et _Metrologia_, 1968, *4*, 45)

[[cipm1967r2r2]]
==== Recommandation 2
```

renders:
![issue24-1](https://user-images.githubusercontent.com/33627611/190491873-040e0dad-8025-4b60-bf96-2b0488e907d5.PNG)

Here, *CIPM recommandation 2* should be replaced by *Recommandation 2 du CIPM*

Same goes with *CCTF Recommandation 2*. It should be replaced by *Recommandation 2 du CCTF*.

In summary, these are the cases that we must address:

* *CIPM recommandation 2* should be *Recommandation 2 du CIPM*
* *CCTF Recommandation 2* should be *Recommandation 2 du CCTF*
* *CGPM Résolution 1* should be *Résolution 1 de la CGPM*
   (CGPM Résolution xx = Résolution xx de la CGPM). Same for CIPM.
* *CGPM Réunion 9* should be *9<sup>e</sup> réunion de la CGPM*
* *CIPM Réunion 51* should be *51<sup>e</sup> réunion du CIPM*

**Comment by ronaldtse** [comment]
This ticket is dependent on:
* https://github.com/metanorma/bipm-si-brochure/issues/193
* https://github.com/relaton/relaton-bipm/issues/34

**Comment by opoudjis** [comment]
I will take on the ticket, but it should just flow through as a result of the relaton updates.

**Comment by opoudjis** [comment]
@ronaldtse The wording of https://github.com/relaton/relaton-bipm/issues/34 indicates that "Recommandation 2 du CIPM" is not to be used, and "CIPM -- Recommandation 2 (1967)" is to be used instead. If that is the case, then both this issue and https://github.com/metanorma/bipm-si-brochure/issues/193 are completely misleading, since what they're requesting is out of date—and in fact, the rendering they will end up with is pretty much what there is now already.

So really, both this issue and https://github.com/metanorma/bipm-si-brochure/issues/193 are obsoleted by https://github.com/metanorma/metanorma-bipm/issues/216 , and should be closed.

@ronaldtse Kindly confirm.

**Comment by ronaldtse** [comment]
This issue is pending on BIPM feedback here:
* https://github.com/metanorma/bipm-si-brochure/pull/210

Please keep this on hold for now.

**Comment by ronaldtse** [comment]
@opoudjis @andrew2net BIPM has approved the format documented in https://github.com/metanorma/bipm-si-brochure/pull/210 , which is now merged into this repository. It's a go!

**Comment by ronaldtse** [comment]
The document to follow is as attached.
[bipm-outcomes-en.pdf](https://github.com/metanorma/metanorma-bipm/files/10671331/bipm-outcomes-en.pdf)

**Comment by opoudjis** [comment]
Awaiting relaton work

**Comment by opoudjis** [comment]
Now awaiting https://github.com/relaton/relaton-bipm/issues/50

**Comment by ronaldtse** [comment]
> @ronaldtse As far as I can tell, the long French citation, _Résolution 1 de la CGPM (1889)_, has **not** been implemented, and has not been provided for in the Brochure. I assume this is not a problem any more?

@andrew2net can you help answer this? We do need this reference. Thanks.

**Comment by ronaldtse** [comment]
@andrew2net when using the language independent identifier:
```sh
$ bundle exec relaton fetch "CGPM RES 1 (1889)"
[relaton-bipm] ("CGPM RES 1 (1889)") fetching...
[relaton-bipm] ("CGPM RES 1 (1889)") not found.
No matching bibliographic entry found
```

**Comment by opoudjis** [comment]
These options are presumably to be realised as alternate docidentifiers, each with its own type. In order to permit the user to specify one docidentifier over another in a citation, I am adding the attribute `style` to erefs. (It is already realised for xrefs). The attribute value for style includes the docidentifier type to be used. It is not limited to that, and future implementations may realise more values.

(This of course is the kind of feature bloat that continues to make Metanorma unlearnable and unmanageable, and I am introducing this feature bloat because Metanorma policy is to provide flexibility to SDOs.)

So assume ref1 = CGPM RES 1, which has a bibitem that looks like:

`````
....
<docidentifier type="BIPM" primary="true">CGPM Resolution (1889)</docidentifier>  
<docidentifier type="BIPM" primary="true" language="fr" script="Latn">CGPM Résolution (1889)</docidentifier>
<docidentifier type="BIPM-long" primary="true" language="fr" script="Latn">Résolution 1&#x1d49; du CGPM (1889)</docidentifier>
`````

Note that I'd rather not introduce formatting into docidentifier, such as:

````
<docidentifier type="BIPM-long" primary="true" language="fr" script="Latn">Résolution 1<sup>e</sup> du CGPM (1889)</docidentifier>
````

Then

* `<<ref1>>` in a French document will automatically resolve to the primary (BIPM) identifier in the right language, `CGPM Résolution (1889)`.
* `<<ref1,style=BIPM-long%>>` will resolve instead to `Résolution 1&#x1d49; du CGPM (1889)`

**Comment by opoudjis** [comment]
It turns out that text citations use the long identifier (which we now call just BIPM for that reason); but the references listing uses the short identifier. I'm coming up with the rule:

  * In other flavours, BIPM documents are both cited and referenced with the BIPM identifier
  * In BIPM itself, documents are referenced (in references) with the BIPM identifier, but cited (in text) with the long identifier.

I'm not obeying that distinction outside of BIPM, outside of BIPM we will just use the one identifier consistently. I will realise BIPM long citations within BIPM documents as a code override (at least for now).

**Comment by opoudjis** [comment]
<img width="783" alt="Screenshot 2023-05-05 at 12 17 04" src="https://user-images.githubusercontent.com/1911039/236420732-74e7056a-68af-416f-b824-56eed362f90b.png">

The precise long French identifier will be debugged. The citation with both short and long French identifier will be fixed when @andrew2net removes the "primary" tag from the long French identifier. This will result in citation by long, referencing by short identifier.
